frontend/src/metabase-lib/expressions/config.js

```diff
@@ -367,6 +367,12 @@ export const MBQL_CLAUSES = {
     type: "number",
     args: ["expression"],
   },
+  "datetime-diff": {
+    displayName: `datetimeDiff`,
+    type: "number",
+    args: ["expression", "expression", "string"],
```

Comment:
this is not really part of this PR but does this mean you the FE lets you put non-temporal stuff in there? Maybe the expression editor should have the notion of arithmetic expressions vs temporal expressions vs string expressions etc.